### ISSUE TITLE
3.15.0 regression: EccKey curve name is inconsistent

### ISSUE THREAD

**Issue by Akuli** [Legrandin/pycryptodome#658]
```
from Crypto.PublicKey import ECC

key = ECC.generate(curve='p384')
print(key)
print(key.curve)
print(key.public_key().curve)
```

pycryptodome 3.14.1 uses `NIST P-384` consistently:

```
EccKey(curve='NIST P-384', ...)
NIST P-384
NIST P-384
```

pycryptodome 3.15.0 sometimes uses `p384` and sometimes `NIST P-384`:

```
EccKey(curve='NIST P-384', ...)
p384
NIST P-384
```

**Comment by Akuli** [comment]
The problem is that 764c1e81 changed `self.curve = self._curve.desc` to `self.curve = curve_name` in `EccKey.__init__`. So now it assigns `self.curve` to whatever is passed as an argument, which accepts multiple different aliases for the same curve. Previously it looked up one "true" curve name from an underlying curve object.

**Comment by Legrandin** [comment]
Thanks for flagging. Fixed with 206d0a8b566941e501a95d26a357b75f1cfe81cf

**Comment by Legrandin** [comment]
Fixed in v3.17
